src/Analysis/Ast/Impl/Modules/Resolution/MainModuleResolution.cs

```diff
@@ -0,0 +1,193 @@
+﻿// Copyright(c) Microsoft Corporation
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the License); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY
+// IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABILITY OR NON-INFRINGEMENT.
+//
+// See the Apache Version 2.0 License for specific language governing
+// permissions and limitations under the License.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Python.Analysis.Core.DependencyResolution;
+using Microsoft.Python.Analysis.Core.Interpreter;
+using Microsoft.Python.Analysis.Documents;
+using Microsoft.Python.Analysis.Types;
+using Microsoft.Python.Core;
+
+namespace Microsoft.Python.Analysis.Modules.Resolution {
+    internal sealed class MainModuleResolution : ModuleResolutionBase, IModuleManagement {
+        private readonly ConcurrentDictionary<string, IPythonModule> _specialized = new ConcurrentDictionary<string, IPythonModule>();
+        private IReadOnlyList<string> _searchPaths;
+
+        public MainModuleResolution(string root, IServiceContainer services)
+            : base(root, services) { }
+
+        internal async Task InitializeAsync(CancellationToken cancellationToken = default) {
+            // Add names from search paths
+            await ReloadAsync(cancellationToken);
+            cancellationToken.ThrowIfCancellationRequested();
+
+            // Initialize built-in
+            var moduleName = BuiltinTypeId.Unknown.GetModuleName(_interpreter.LanguageVersion);
+            var modulePath = ModuleCache.GetCacheFilePath(_interpreter.Configuration.InterpreterPath ?? "python.exe");
```

Comment:
Is `python.exe` actually what the default should be? I can see that it's a carryover from `AstBuiltinsPythonModule.cs`, it just seems strange.

Reply:
We should drop default. Either real one, or out :-)

Reply:
Remove in upcoming PR